### ISSUE TITLE
pods can get stuck Terminating in certain situations

### ISSUE THREAD

**Issue by DenisIzmaylov** [kubernetes/kubernetes#27013]
Extracted from #24819. When I configure and do test deploy with `my-nginx` it will produce the following output for `kubectl get pods`:

```
...
my-nginx-3869474375-oj1i0    0/1       Terminating   0          20h
my-nginx-3869474375-oui1w    0/1       Terminating   0          20h
my-nginx-3869474375-p0ixa    0/1       Terminating   0          20h
my-nginx-3869474375-p21aq    0/1       Terminating   0          20h
my-nginx-3869474375-p3e3q    0/1       Terminating   0          20h
my-nginx-3869474375-p5qxp    0/1       Terminating   0          20h
my-nginx-3869474375-pcesx    0/1       Terminating   0          20h
my-nginx-3869474375-q238e    0/1       Terminating   0          20h
my-nginx-3869474375-qto18    0/1       Terminating   0          20h
my-nginx-3869474375-r4u6b    0/1       Terminating   0          20h
my-nginx-3869474375-rlv4a    0/1       Terminating   0          20h
my-nginx-3869474375-rucy2    0/1       Terminating   0          20h
my-nginx-3869474375-rvmqo    0/1       Terminating   0          20h
my-nginx-3869474375-ry7bj    0/1       Terminating   0          20h
my-nginx-3869474375-s2mwi    0/1       Terminating   0          20h
...
```

When I run `kubectl delete pods --grace-period=0 --all` to remove all pods (it's a new cluster so it's OK) it will clear the list. But in few seconds later they are returned back. 

Details about a sample pod:
```
Name:				my-nginx-3869474375-zygrz
Namespace:			default
Node:				<removed IP>
Labels:				pod-template-hash=3869474375,run=my-nginx
Status:				Terminating (expires Tue, 07 Jun 2016 20:03:08 -0400)
Termination Grace Period:	30s
IP:				
Controllers:			ReplicaSet/my-nginx-3869474375
Containers:
  my-nginx:
    Image:	nginx
    Port:	81/TCP
    QoS Tier:
      cpu:	BestEffort
      memory:	BestEffort
    Environment Variables:
Volumes:
  default-token-i7wig:
    Type:	Secret (a volume populated by a Secret)
    SecretName:	default-token-i7wig
Events:
  FirstSeen	LastSeen	Count	From			SubobjectPath	Type		Reason			Message
  ---------	--------	-----	----			-------------	--------	------			-------
  1m		1m		1	{default-scheduler }			Normal		Scheduled		Successfully assigned my-nginx-3869474375-zygrz to <removed IP>
  1m		1m		1	{kubelet <removed IP>}			Warning		MissingClusterDNS	kubelet does not have ClusterDNS IP configured and cannot create Pod using "ClusterFirst" policy. Falling back to DNSDefault policy.
```

**Comment by DenisIzmaylov** [comment]
@yujuhong asked:

> How long did your pods stay in the terminating state? Have they ever become running before you attempted to delete them?

I've just removed all but they started again so it's running (stucked) a few minutes.

**Comment by DenisIzmaylov** [comment]
Some of them could be `pending` for a few seconds:
```
my-nginx-3869474375-aiijr    0/1       Terminating   0          2m
my-nginx-3869474375-ausff    0/1       Pending       0          27m
my-nginx-3869474375-avxzf    0/1       Terminating   0          1m
```

**Comment by yujuhong** [comment]
@DenisIzmaylov, you should post this on [stackoverflow](http://stackoverflow.com/questions/tagged/kubernetes), where the community (i.e., more users) can help you troubleshoot the issue. Alternatively, you can go on our [slack channel](https://kubernetes.slack.com/messages/kubernetes-users/) to ask for help.

> I've just removed all but they started again so it's running (stucked) a few minutes.

I meant has kubelet ever truly started your containers and report the pod status as running and ready?
In other words, if you just create a new pod, wait for a while (5min?) and don't try to delete the pod, what will be the output of the pod status?

I am going to close this issue for now, but feel free to post the link to the stackoverflow question you created.

**Comment by DenisIzmaylov** [comment]
Thank you. 

I've just deleted all pods but in 20 minutes I got all they came back:
```
root@startup-makers:~# kubectl get pods
NAME                         READY     STATUS        RESTARTS   AGE
my-nginx-3869474375-0e5wp    0/1       Terminating   0          17m
my-nginx-3869474375-0v4lp    0/1       Terminating   0          18m
my-nginx-3869474375-2ish6    0/1       Terminating   0          18m
my-nginx-3869474375-3u1mi    0/1       Terminating   0          18m
my-nginx-3869474375-5hg7z    0/1       Terminating   0          18m
my-nginx-3869474375-5ltyr    0/1       Terminating   0          18m
my-nginx-3869474375-6cz4j    0/1       Terminating   0          18m
my-nginx-3869474375-6t8px    0/1       Terminating   0          18m
my-nginx-3869474375-6wwpm    0/1       Terminating   0          18m
my-nginx-3869474375-7od6k    0/1       Terminating   0          17m
my-nginx-3869474375-8qvh0    0/1       Terminating   0          18m
my-nginx-3869474375-9eqzb    0/1       Terminating   0          18m
my-nginx-3869474375-asveo    0/1       Terminating   0          18m
my-nginx-3869474375-ausff    0/1       Terminating   0          53m
my-nginx-3869474375-bx3yb    0/1       Terminating   0          18m
my-nginx-3869474375-euotp    0/1       Terminating   0          18m
my-nginx-3869474375-evxlc    0/1       Terminating   0          18m
my-nginx-3869474375-gwihj    0/1       Terminating   0          18m
my-nginx-3869474375-j6wgm    0/1       Terminating   0          18m
my-nginx-3869474375-jcgcd    0/1       Terminating   0          18m
my-nginx-3869474375-jegdh    0/1       Terminating   0          18m
my-nginx-3869474375-kthtb    0/1       Terminating   0          18m
my-nginx-3869474375-m7n2f    0/1       Terminating   0          18m
my-nginx-3869474375-n4kd7    0/1       Terminating   0          18m
my-nginx-3869474375-oij5c    0/1       Terminating   0          18m
my-nginx-3869474375-pijip    0/1       Terminating   0          18m
my-nginx-3869474375-r48v2    0/1       Terminating   0          18m
my-nginx-3869474375-sdugu    0/1       Terminating   0          18m
my-nginx-3869474375-spwym    0/1       Terminating   0          18m
my-nginx-3869474375-syh4l    0/1       Terminating   0          18m
my-nginx-3869474375-tdwl0    0/1       Terminating   0          18m
my-nginx-3869474375-u8nga    0/1       Terminating   0          18m
my-nginx-3869474375-v8h2c    0/1       Terminating   0          18m
my-nginx-3869474375-w077v    0/1       Terminating   0          18m
my-nginx-3869474375-x5k46    0/1       Terminating   0          18m
my-nginx-3869474375-x9gii    0/1       Terminating   0          18m
my-nginx-3869474375-ydu0g    0/1       Terminating   0          17m
my-nginx-3869474375-zug3c    0/1       Terminating   0          18m
my-nginx-aekm1               1/1       Running       0          6m
my-nginx-nmgu8               1/1       Running       0          6m
my-nginx2-1054506668-08g3l   0/1       Terminating   0          52m
```

How I can stop it? Which is service can do it?

**Comment by DenisIzmaylov** [comment]
Oh, ok. I deleted services and replicas and then delete all pods again:
```
kubectl delete pods --all --grace-period=0
```
Now all pods is disappeared as it estimated.

**Comment by therc** [comment]
I found an example:
```2016-10-07 16:54:03 +0000 UTC   2016-10-07 16:49:03 +0000 UTC   6         XYZ-908500842-z9aa7   Pod                 Warning   FailedValidation   {kubelet ip-10-ABC}   Error validating pod XYZ-908500842-z9aa7.staging from api, ignoring: spec.containers[0].securityContext.privileged: Forbidden: disallowed by policy```

It apparently never ran and was in Terminating state for 14h. `--grace-period=0` was the only way to get rid of it.
